api/app/clients/GoogleClient.js

```diff
@@ -717,6 +727,123 @@ class GoogleClient extends BaseClient {
     return reply;
   }
 
+  /**
+   * Stripped-down logic for generating a title. This uses the non-streaming APIs, since the user does not see titles streaming
+   */
+  async titleChatCompletion(_payload, options = {}) {
```

Comment:
Just a question here, is it compatible with all the different google clients?

Reply:
Yes, all that support text-only.

It's a copy of the `getCompletion` method, but with a non-streaming call and no vision use cases (since Google's vision models fail if an image isn't provided).